packages/plugin-core/src/test/suite-integ/CapitalizeText.test.ts

```diff
@@ -0,0 +1,123 @@
+import { ENGINE_HOOKS } from "@dendronhq/common-test-utils";
+import _ from "lodash";
+import { ExtensionContext, Position, Selection } from "vscode";
+import { CapitalizeTextCommand } from "../../commands/CapitalizeText";
+import { VSCodeUtils } from "../../utils";
+import { expect } from "../testUtilsv2";
+import { runLegacyMultiWorkspaceTest, setupBeforeAfter } from "../testUtilsV3";
+
+suite("notes", function () {
+  let ctx: ExtensionContext;
+
+  ctx = setupBeforeAfter(this, {});
+
+  test("with no selections", (done) => {
+    runLegacyMultiWorkspaceTest({
+      ctx,
+      preSetupHook: ENGINE_HOOKS.setupBasic,
+      onInit: async ({ vaults }) => {
+        const vault = vaults[0];
+        const fname = "foo";
+
+        const editor = await VSCodeUtils.openNoteByPath({ vault, fname });
+
+        const lowercaseText = "undercase text";
+
+        const start = new Position(8, 0);
+        const end = new Position(8, lowercaseText.length);
+        const selection = new Selection(start, end);
+
+        await editor.edit((editBuilder) => {
+          editBuilder.insert(start, lowercaseText);
+        });
+
+        await new CapitalizeTextCommand().run();
+
+        const changedText = editor.document.getText(selection);
+
+        expect(changedText).toEqual(lowercaseText);
+        done();
+      },
+    });
+  });
+
+  test("with one selection", (done) => {
+    runLegacyMultiWorkspaceTest({
+      ctx,
+      preSetupHook: ENGINE_HOOKS.setupBasic,
+      onInit: async ({ vaults }) => {
+        const vault = vaults[0];
+        const fname = "foo";
+
+        const editor = await VSCodeUtils.openNoteByPath({ vault, fname });
+
+        const lowercaseText = "undercase text";
+        const uppercaseText = "UNDERCASE TEXT";
+
+        const start = new Position(8, 0);
+        const end = new Position(8, lowercaseText.length);
+        const selection = new Selection(start, end);
+
+        await editor.edit((editBuilder) => {
+          editBuilder.insert(start, lowercaseText);
+        });
+
+        editor.selection = selection;
+
+        await new CapitalizeTextCommand().run();
+
+        const changedText = editor.document.getText(selection);
+
+        expect(changedText).toEqual(uppercaseText);
+        done();
+      },
+    });
+  });
+
+  test("with multiple selections", (done) => {
```

Comment:
nice! not many people would think to test for multiple selections :)

Reply:
Thanks! I use multiple selections in VSCode frequently, so it seemed like a natural use case to consider.